analyzers/src/SonarAnalyzer.Common/Rules/ParameterNameMatchesOriginalBase.cs

```diff
@@ -64,7 +65,7 @@ public abstract class ParameterNameMatchesOriginalBase<TSyntaxKind, TMethodDecla
                         }
                         else if (methodSymbol.GetInterfaceMember() is { } interfaceMember)
                         {
-                            VerifyParameters(c, methodSyntax, interfaceMember.Parameters, "interface");
+                            VerifyInterfaceMemberParameters(c, methodSymbol, methodSyntax, interfaceMember, "interface");
```

Comment:
As we discussed in an offline talk, let`s try to cover the following cases:

```
    public abstract class BaseClass<T>
    {
        public abstract void SomeMethod(T someParameter);
    }

    public class ClassOne : BaseClass<int>
    {
        public override void SomeMethod(int renamedParam)
        {
        }
    }

    public interface IGenericInterface<T>
    {
        void SomeMethod(T someParameter);
    }

    public class Implementation : IGenericInterface<int>
    {
        public void SomeMethod(int renamedParam)
        {
        }
    }
```

And as a bonus if it is a low hanging fruit :

  ```
  public abstract record BaseRecord<T>
    {
        public abstract void SomeMethod(T someParameter);
    }

    public record RecordOne : BaseRecord<int>
    {
        public override void SomeMethod(int renamedParam)
        {
        }
    }
```